src/NuGet.Clients/NuGet.VisualStudio.Internal.Contracts/ContextInfos/PackageSearchMetadataContextInfo.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using MessagePack;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.VisualStudio.Internal.Contracts
+{
+    [MessagePackObject(keyAsPropertyName: true)]
+    public sealed class PackageSearchMetadataContextInfo
+    {
+        public PackageIdentity? Identity { get; set; }
+        public string? Title { get; set; }
+        public string? Description { get; set; }
+        public string? Authors { get; set; }
+        public Uri? IconUrl { get; set; }
+        public string? Tags { get; set; }
+        public Uri? LicenseUrl { get; set; }
+        public string? Owners { get; set; }
+        public Uri? ProjectUrl { get; set; }
+        public DateTimeOffset? Published { get; set; }
+        public Uri? ReportAbuseUrl { get; set; }
+        public Uri? PackageDetailsUrl { get; set; }
+        public bool RequireLicenseAcceptance { get; set; }
+        public string? Summary { get; set; }
+        public bool PrefixReserved { get; set; }
+        public bool IsRecommended { get; set; }
+        public (string modelVersion, string vsixVersion)? RecommenderVersion { get; set; }
+        public bool IsListed { get; set; }
+        public long? DownloadCount { get; set; }
+        public IEnumerable<PackageDependencyGroup>? DependencySets { get; set; }
```

Comment:
Use `IReadOnlyCollection<T>` or `IReadOnlyList<T>` instead of `IEnumerable<T>` in serializable types.  This feedback applies to all message packable types.